### PR TITLE
Fix issues relating to past or completed Events and NOK parsers

### PR DESCRIPTION
**Pull request by ntwbruce**
Fixes:
- Possible to add Events in the past
    - add method in EventDate to check if given date is in the present/future
    - add guard clause in parseEventDate in ParserUtil (which is used in AddEventCommand) to throw exception if given date is past
- Possible to add Participant to completed Event
    - add guard clauses in AddParticipantToEventCommand to throw exceptions if Event is completed or in the past
- Possible to add Participant to Event before date of birth
    - prevented by preventing adding of Events in the past (added above) and adding of Birthdates in the future (existing)
- Displaying a single Event returns "1 events listed"
    - change message to "event(s) listed"
- addNok and deleteNok commands cannot handle extra whitespaces
    - modify split regex to handle whitespaces
- Misc
    - Update MESSAGE_SUCCESS for NOK classes
    - Update EventCard
    - Update test classes (mostly changing all the sample dates to in the future)

### CODE REVIEW

**Review by XXJJXJ — COMMENTED** [review]
LGTM, but do we need to check for events in the past? (I don't think it is required, but it doesn't matter too I guess, but if yall want, then some of our event's sample data might need to change)(Im not sure if those will cause any errors)

Will take a closer look after tomorrow

**Review by XXJJXJ — APPROVED** [review]
Actually now I think of it, perhaps we should let past events to be created too (maybe can put in UG to say, for archival/recording purposes). Since this will cause our testcases to fail overtime (abit weak in maintainability). Because, what I did was I also tried example commands in UG and it would fail if the dates aren't up to date. However, the preventing participants from enrolling to past events can be kept. wdyt?

File: src/main/java/seedu/address/ui/EventCard.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public EventCard(Event event, int displayedIndex) {
         date.setText(event.getDateString());
         time.setText(event.getTimeDisplayString());
         int noOfParticipants = event.getParticipants().size();
-        if (noOfParticipants <= 1) {
+        if (noOfParticipants == 1) {
```
Comment (XXJJXJ) [inline src/main/java/seedu/address/ui/EventCard.java:42]: Actly this should be <= 1 I guess, cuz 0 participant also?

Comment (ntwbruce) [reply src/main/java/seedu/address/ui/EventCard.java:42]: I think it should be "0 participants", grammatically speaking

Comment (XXJJXJ) [reply src/main/java/seedu/address/ui/EventCard.java:42]: Oh, then this ba, my ang moh not strong enough 😅

**Comment by ntwbruce** [comment]
Closing PR because cannot add to v1.4

**Pull request closed without merging**